crypto/signature.go

```diff
@@ -81,20 +89,19 @@ func processSignatureExpiration(md *openpgp.MessageDetails, verifyTime int64) {
 
 // verifyDetailsSignature verifies signature from message details.
 func verifyDetailsSignature(md *openpgp.MessageDetails, verifierKey *KeyRing) error {
-	if md.IsSigned {
-		if md.SignedBy == nil || len(verifierKey.entities) == 0 {
-			return newSignatureNoVerifier()
-		}
-		matches := verifierKey.entities.KeysById(md.SignedByKeyId)
-		if len(matches) > 0 {
-			if md.SignatureError == nil {
-				return nil
-			}
-			return newSignatureFailed()
-		}
+	if !md.IsSigned ||
+		md.SignedBy == nil ||
+		len(verifierKey.entities) == 0 ||
+		len(verifierKey.entities.KeysById(md.SignedByKeyId)) == 0 {
+		return newSignatureNoVerifier()
 	}
-
-	return newSignatureNoVerifier()
+	if md.SignatureError != nil ||
```

Comment:
Maybe an `&&` for extra security? Or is there a case where a `SignatureError` is given on a valid signature?

Reply:
The body of this if statement returns an error, so it errors if there is a `SignatureError` no matter what else happens.